Add assignment-7

antriksh-webonise opened the pull request:
- CRUD operations using Spring MVC
-JAVA-config based using JDBC

antriksh-webonise committed:
Add assignment-7
- CRUD operations using Spring MVC
-JAVA-config based using JDBC

shrikant-webonise commented on EmployeePortal using SpringMVC(JAVA config based using JDBC)/src/main/java/com/employeeportal/config/ServletInitializer.java:
```diff
@@ -0,0 +1,47 @@
+    @Bean
+    public DataSource dataSource() {
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
+        dataSource.setDriverClassName("com.mysql.jdbc.Driver");
```
Extract this into a constant.

shrikant-webonise commented on EmployeePortal using SpringMVC(JAVA config based using JDBC)/src/main/java/com/employeeportal/config/ServletInitializer.java:
```diff
@@ -0,0 +1,47 @@
+        dataSource.setDriverClassName("com.mysql.jdbc.Driver");
+        dataSource.setUrl("jdbc:mysql://localhost:3306/Organization");
+        dataSource.setUsername("root");
+        dataSource.setPassword("Spacex@123");
```
Same as above.

shrikant-webonise commented on EmployeePortal using SpringMVC(JAVA config based using JDBC)/src/main/java/com/employeeportal/controllers/EmployeeController.java:
```diff
@@ -0,0 +1,55 @@
+        return new ModelAndView("redirect:/viewemp");
+    }
+
+    @RequestMapping(value="/deleteemp/{id}",method = RequestMethod.GET)
```
It should have a RequestMethod.POST or RequestMethod.DELETE type.

shrikant-webonise commented on EmployeePortal using SpringMVC(JAVA config based using JDBC)/src/main/java/com/employeeportal/controllers/EmployeeController.java:
```diff
@@ -0,0 +1,55 @@
+        return modelAndView;
+    }
+
+    @RequestMapping(value="/editemp/{id}")
```
You can use "/employee/{id}" as a request mapping and remove "edit" as it gives you the employee details, however "/employee" can be extracted outside the class.

shrikant-webonise commented on EmployeePortal using SpringMVC(JAVA config based using JDBC)/src/main/java/com/employeeportal/model/Employee.java:
```diff
@@ -0,0 +1,50 @@
+package com.employeeportal.model;
+
```
Please document the model class.

shrikant-webonise commented on EmployeePortal using SpringMVC(JAVA config based using JDBC)/src/main/java/com/employeeportal/service/EmployeeService.java:
```diff
@@ -0,0 +1,18 @@
+
+public interface EmployeeService {
+
+    public int save(Employee employee) throws Exception;
```
Please document all the interface methods.

antriksh-webonise committed:
Make the changes
- Add the documentation to model class and interfaces
- Extract the connection parameters to a constant

antriksh-webonise committed:
Delete gitignore
